protocol/carbon/carbon.go

```diff
@@ -40,9 +40,14 @@ func NewCarbonDatapoint(line string, metricDeconstructor metricdeconstructor.Met
 	if err != nil {
 		return nil, err
 	}
-	metricTime, err := strconv.ParseInt(parts[2], 10, 64)
-	if err != nil {
-		return nil, errors.Annotatef(err, "invalid carbon metric time on input line %s", line)
+
+	var metricTime int64
+	if metricTime, err = strconv.ParseInt(parts[2], 10, 64); err != nil {
```

Comment:
All integer values can be parsed as a float with exact precision so this differentiation in parsing is unnecessary.